contracts/ThalesRoyale/ThalesRoyale.sol

```diff
@@ -316,50 +369,71 @@ contract ThalesRoyale is Initializable, ProxyOwned, PausableUpgradeable, ProxyRe
     /* ========== INTERNALS ========== */
 
     function _signUpPlayer(address _player, uint _position) internal {
+        uint tokenId = thalesRoyalePassport.safeMint(_player);
+        tokenSeason[tokenId] = season;
+
+        tokensMintedPerSeason[season][tokenId] = block.timestamp;
+        tokensPerSeason[season].push(tokenId);
+        mintedTokensCount[season]++;
+
         playerSignedUpPerSeason[season][_player] = block.timestamp;
-        playersPerSeason[season].push(_player);
-        signedUpPlayersCount[season]++;
 
         if (_position != 0) {
             for(uint i = 1; i <= rounds; i++){
-                _putPosition(_player, season, i, _position);
+                _putPosition(_player, season, i, _position, tokenId);
             }
         }
 
         _buyIn(_player, buyInAmount);
 
-        emit SignedUp(_player, season, _position);
+        emit SignedUp(_player, tokenId, season, _position);
     }
 
     function _signUpPlayerWithPass(
         address _player,
         uint _position,
         uint _passId
     ) internal {
+        uint tokenId = thalesRoyalePassport.safeMint(_player);
+        tokenSeason[tokenId] = season;
+
+        tokensMintedPerSeason[season][tokenId] = block.timestamp;
+        tokensPerSeason[season].push(tokenId);
+        mintedTokensCount[season]++;
+
         playerSignedUpPerSeason[season][_player] = block.timestamp;
-        playersPerSeason[season].push(_player);
-        signedUpPlayersCount[season]++;
+
 
         if (_position != 0) {
             for(uint i = 1; i <= rounds; i++){
```

Comment:
i know it is mine part of a code but i see it now it is repeated in line 382 can you refactor that 🙏  sorry